sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/DataLakeDirectoryAsyncClient.java

```diff
@@ -485,6 +496,63 @@ public DataLakeDirectoryAsyncClient getSubdirectoryAsyncClient(String subdirecto
         }
     }
 
+    /**
+     * Returns a reactive Publisher emitting all the files/directories in this account lazily as needed. For more
+     * information, see the <a href="https://docs.microsoft.com/en-us/rest/api/storageservices/datalakestoragegen2/filesystem/list">Azure Docs</a>.
+     *
+     * <p><strong>Code Samples</strong></p>
+     *
+     * {@codesnippet com.azure.storage.file.datalake.DataLakeFileSystemAsyncClient.listPaths}
+     *
+     * @return A reactive response emitting the list of files/directories.
+     */
+    public PagedFlux<PathItem> listPaths() {
+        try {
+            return this.listPaths(false, false, null);
+        } catch (RuntimeException ex) {
```

Comment:
Should move this try/catch deeper into the call stack, either the public overload or in the package-private method so that we don't need to put it in multiple locations.